From 069778976dfdd1d54e39a136d587a0174b089c35 Mon Sep 17 00:00:00 2001
From: Murali Karicheri <m-karicheri2@ti.com>
Date: Tue, 2 Jan 2018 12:59:01 -0500
Subject: [PATCH 150/153] prueth: fix the kernel crash when eth interface is
 brought down.

When Ethernet interface is brought down using ifconfig down command
following kernel crash dump is shown at the console.

root@am57xx-evm:~# ifconfig eth3 0.0.0.0 down
[  703.238601] Unable to handle kernel NULL pointer dereference at
 virtual address 00000090
[  703.246732] pgd = d53a4d40
[  703.249663] [00000090] *pgd=95302003, *pmd=fb8b9003
[  703.254585] Internal error: Oops: 207 [#1] PREEMPT SMP ARM
[  703.254678] Modules linked in: bc_example(O) xhci_plat_hcd xhci_hcd
usbcore rpmsg_proto rpmsg_rpc dwc3 udc_core usb_common sha512_generic
sha512_arm sha256_g)
[  703.254685] CPU: 0 PID: 1184 Comm: ifconfig Tainted:
 G           O    4.9.59-rt23-g273202a0d3 #2
[  703.254687] Hardware name: Generic DRA74X (Flattened Device Tree)
[  703.254690] task: d50b87c0 task.stack: ee64e000
[  703.254700] PC is at rt_mutex_lock_state+0x24/0x5c
[  703.254706] LR is at __down_write_common+0x20/0x13c
[  703.254711] pc : [<c08be890>]    lr : [<c08bea70>]    psr: 80070013
[  703.254711] sp : ee64fd10  ip : d50b87c0  fp : ee64fd1c
[  703.254714] r10: 00000000  r9 : 00000000  r8 : 00000002
[  703.254717] r7 : ed774a6c  r6 : ed4c8580  r5 : 00000084  r4 : 00000080
[  703.254720] r3 : ee64e000  r2 : 00000090  r1 : 00000002  r0 : 00000084
[  703.254724] Flags: Nzcv  IRQs on  FIQs on  Mode SVC_32  ISA ARM
 Segment user

Signed-off-by: Murali Karicheri <m-karicheri2@ti.com>
---
 drivers/net/ethernet/ti/prueth.c | 3 ---
 1 file changed, 3 deletions(-)

diff --git a/drivers/net/ethernet/ti/prueth.c b/drivers/net/ethernet/ti/prueth.c
index 0958214..5060881 100644
--- a/drivers/net/ethernet/ti/prueth.c
+++ b/drivers/net/ethernet/ti/prueth.c
@@ -1112,10 +1112,7 @@ int prueth_hsr_prp_debugfs_init(struct prueth *prueth)
 	prueth->tr_file = NULL;
 	prueth->error_stats_file = NULL;
 	prueth->root_dir = NULL;
-
-	debugfs_remove(prueth->new_nt_index);
 	prueth->new_nt_index = NULL;
-	debugfs_remove(prueth->new_nt_bins);
 	prueth->new_nt_bins = NULL;
 }
 #else
-- 
1.9.1

